src/backend/joanie/core/api/client/__init__.py

```diff
@@ -380,10 +380,31 @@ def _get_organization_with_least_active_orders(
         except models.Organization.DoesNotExist:
             return None
 
-    # pylint: disable=too-many-return-statements
+    # pylint: disable=too-many-return-statements, too-many-locals
     @transaction.atomic
     def create(self, request, *args, **kwargs):
         """Try to create an order and a related payment if the payment is fee."""
+        if voucher_code := request.data.get("voucher_code"):
+            voucher = models.Voucher.objects.get(code=voucher_code)
+            user = models.User.objects.get(username=self.request.auth["username"])
+
+            if not voucher.is_usable_by(user.id):
+                return Response(
+                    f"Voucher already claimed by user {user.id}",
+                    status=HTTPStatus.BAD_REQUEST,
+                )
+
+            order = models.Order.objects.filter(
+                voucher__code=voucher_code, state=enums.ORDER_STATE_TO_OWN
```

Comment:
I wonder if we should check for a rate 1 voucher here…
```suggestion
                voucher__code=voucher_code, voucher__rate=1, state=enums.ORDER_STATE_TO_OWN
```